### DISCUSSION TITLE
How to hook into Save event of CustomSettings Content Type

### DISCUSSION THREAD

**Discussion by kgwadu** [OrchardCMS/OrchardCore#12053] [Q&A] [answered]
Similar to how you can hook into the publish event for content items, we have a requirement to hook into the save / update event for a CustomSettings content type. I have tried to look for an implementation or any information but have had no luck so far.

Would greatly appreciate it if someone could assist me with this or point me in the right direction.

**Comment by jtkech** [comment, marked as answer]
On update it calls `_contentItemDisplayManager.UpdateEditorAsync()`

https://github.com/OrchardCMS/OrchardCore/blob/c255456bf1d9093d13447e17485a23a1253f3e9b/src/OrchardCore.Modules/OrchardCore.CustomSettings/Drivers/CustomSettingsDisplayDriver.cs#L67-L69

which calls 

https://github.com/OrchardCMS/OrchardCore/blob/c255456bf1d9093d13447e17485a23a1253f3e9b/src/OrchardCore/OrchardCore.ContentManagement.Display/ContentItemDisplayManager.cs#L182-L184

So looks like you can still hook on the updating and updated events. Hmm as I remember from a `: IContentHandler` or simpler a `: ContentHandlerBase`, not a `ContentPartHandler`, then from the context you can retrieve the content item on which you can check if it is of your custom settings content type.

**Reply by kgwadu** [reply]
@jtkech appreciate the super prompt reply, will take a look at what you've suggested and see how we go.
